tests/Fixer/ControlStructure/NoEmptyBlockFixerTest.php

```diff
@@ -0,0 +1,815 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests\Fixer\ControlStructure;
+
+use PhpCsFixer\Tests\Test\AbstractFixerTestCase;
+
+/**
+ * @author Nat Zimmermann <nathanielzimmermann@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\ControlStructure\NoEmptyBlockFixer
+ */
+final class NoEmptyBlockFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideTestFixCases
+     */
+    public function testFix($expected, $input = null): void
+    {
+        $this->doTest($expected, $input);
+    }
+
+    public static function provideTestFixCases(): iterable
```

Comment:
```php

<?php

echo 1;

{}

echo 2;
```

Reply:
`Fixer must be a candidate for input code.` - currently fixer works for loops, switches etc. Is it required to handle this case?

Reply:
I would imagine `{}` to be handled by `NoEmptyBlockFixer`, if we want to touch only `loop`, then we would have `NoEmptyBodyBlockOfLoopFixer, but I would recommend to not have that very specialized rule that covers only some limited cases

Reply:
I agree, I just don't think that rule must cover every possible scenario since the very beginning. It's not my code, I just refurbished it from the ancient branch, I would like to provide it "as-is" (not counting required alignments). Then we can improve it even more. Example with `{}` is such an edge case that in my opinion doesn't have to be handled now.